NonEuclidean/Engine.h

```diff
@@ -23,26 +27,46 @@ class Engine {
   void Render(const Camera& cam, GLuint curFBO, const Portal* skipPortal);
   void LoadScene(int ix);
 
+#if defined(_WIN32)
   LRESULT WindowProc(HWND hCurWnd, UINT uMsg, WPARAM wParam, LPARAM lParam);
+#endif
 
   const Player& GetPlayer() const { return *player; }
   float NearestPortalDist() const;
 
 private:
+  bool InitOSWrapper();
   void CreateGLWindow();
+  void DestroyGLWindow();
   void InitGLObjects();
   void DestroyGLObjects();
   void SetupInputs();
   void ConfineCursor();
   void ToggleFullscreen();
+  int EnterMessageLoop();
+  void PeriodicRender(int64_t cur_ticks);
```

Comment:
cur_ticks must be passed by reference here because it's modified in the method.  Without that, it's running at a hyper-speed framerate.